pumpkin-lib/src/engine/cp/assignments_integer.rs

```diff
@@ -28,38 +31,49 @@ pub struct AssignmentsInteger {
     backtrack_events: EventSink,
 }
 
+/// A structure which indicates that an empty domain has been encountered; oftentimes returned as
+/// an [`Err`] variant.
 #[derive(Clone, Copy, Debug)]
 pub struct EmptyDomain;
 
 impl AssignmentsInteger {
+    /// Increments the current decision level
     pub fn increase_decision_level(&mut self) {
         self.trail.increase_decision_level()
     }
 
+    /// Returns the current decision level
     pub fn get_decision_level(&self) -> usize {
         self.trail.get_decision_level()
     }
 
+    /// Returns the number of defined [`DomainId`]s.
     pub fn num_domains(&self) -> u32 {
         self.domains.len() as u32
     }
 
+    /// returns an iterator of all the [`DomainId`]s.
     pub fn get_domains(&self) -> DomainGeneratorIterator {
         DomainGeneratorIterator::new(0, self.num_domains())
     }
 
+    /// Returns the number of entries on the trail.
```

Comment:
Should we make a note that this does not necessarily equal the number of propagations which have occurred (e.g. due to the splitting of equality predicates into two predicate)?